lego/apps/events/serializers/events.py

```diff
@@ -79,6 +80,23 @@ class Meta:
         )
         read_only = True
 
+    def user_should_see_regs(self, event, user):
```

Comment:
Maybe it's possible to move this to the view and pass it to the serializer, so we won't have to call it twice?

Reply:
I think we are better off handling this in the viewset, because this check is done for every pool in the event. This could be solved by just switching serializers. One without pools and registrations, and one with. That way you can just inherit the one without pools and add the pool and waiting_registration fields to minimize duplicate code.

Reply:
Or wait, i think it should only run once - how did you get this to run twice? Anyways, better off selecting different serializers imo.

Reply:
It'll run during `get_pools` to pass an argument to the pool serializer about whether or not to include registrations, and one during `get_waiting_registrations` to check whether to include all waiting regs or to just return the number of waiting regs

Reply:
Would it be ok to include `waiting_registrations_count` and `pool.registration_count` for every event, or only those where registrations are hidden?